websockets/websockets_test.go

```diff
@@ -879,3 +880,241 @@ func TestCookiesDefaultJar(t *testing.T) {
 	assert.Equal(t, map[string]string{"someheader": "defaultjar"}, collected)
 	mu.Unlock()
 }
+
+func TestCompression(t *testing.T) {
+	t.Parallel()
+
+	t.Run("session", func(t *testing.T) {
+		t.Parallel()
+		const text string = `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sed pharetra sapien. Nunc laoreet molestie ante ac gravida. Etiam interdum dui viverra posuere egestas. Pellentesque at dolor tristique, mattis turpis eget, commodo purus. Nunc orci aliquam.`
+
+		ts := newTestState(t)
+		sr := ts.tb.Replacer.Replace
+		ts.tb.Mux.HandleFunc("/ws-compression", http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+			upgrader := websocket.Upgrader{
+				EnableCompression: true,
+				ReadBufferSize:    1024,
+				WriteBufferSize:   1024,
+			}
+
+			conn, e := upgrader.Upgrade(w, req, w.Header())
+			if e != nil {
+				t.Fatalf("/ws-compression cannot upgrade request: %v", e)
```

Comment:
Calling (and in general using) `testing.T` of the goroutine that started the thread is considered "bad", and may or may not *work*

You can see this (somewhat) more detailed on https://pkg.go.dev/testing#T. 

In general this is as there is nothing stopping you from trying to fail after the test has returned (and has been marked failed). 

Usually in this cases the error gets reported through a (buffered) channel (or some other method) and the main thread then checks that no error happened.

This usually isn't really fatal as the error will still get reported just go will also tell you you did use testing.T off the main thread potentially mudding the error.

Also unfortunately it really commonly happens (to me at least) to write this code and not realize it. I am pretty sure there is a lot of this in k6 core that we have not fixed :(.

Reply:
Sure, thing.

I'll adjust all tests.